src/solidity-frontend/solidity_convert.cpp

```diff
@@ -2274,12 +2272,13 @@ bool solidity_convertert::get_binary_operator_expr(
   }
   else if (expr.contains("leftExpression"))
   {
-    nlohmann::json commonType = expr["commonType"];
+    nlohmann::json literalType_l = expr["leftExpression"]["typeDescriptions"];
+    nlohmann::json literalType_r = expr["rightExpression"]["typeDescriptions"];
 
-    if (get_expr(expr["leftExpression"], commonType, lhs))
+    if (get_expr(expr["leftExpression"], literalType_l, lhs))
```

Comment:
I didn't check the differences in the implementations of the two get_expr() versions, one with a `common_type` parameter and one without it. But to me it seems that this change (and also the `"rightExpression"` just below) are in fact not using a "common type" anymore, but the type of whatever expression is already passed to get_expr(). Maybe these should call the 2-parameter version without the common type?

Reply:
I just tested the two-parameter version and it failed almost every regression test. I believe we still need the type information (`literalType`) when parsing the literal.

Reply:
Alright. Thanks for checking. This might mean, thought, that the type parameter, which is currently named `expr_common_type` is not really a "common" type. But never mind, this is independent of this PR.